src/app/services/sound.service.ts

```diff
@@ -36,4 +30,16 @@ export class SoundService {
   toggleMute() {
     this.isMuted = !this.isMuted;
   }
+
+  private getSound(sound: Sound): HTMLAudioElement {
+    // look for the loaded sound in cache
+    let element = this.loadedSounds[sound];
+    if (element) return element;
+
+    // otherwise load the file, cache it, and return it
+    element = new Audio(`${BASE_PATH}/${sound}`);
+    element.load();
```

Comment:
If we wait to load the sound until the first time that it's played, and then cache it for later use, we don't need to load all of the sounds at the beginning. We could load everything at the beginning though if we want to speed up the first play of the sound. What do you think? I don't know the performance implications.